spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -103,8 +105,16 @@ public CompletableFuture<MessageId> sendAsync(String topic, T message,
 
 	private Producer<T> prepareProducerForSend(String topic, T message, MessageRouter messageRouter)
 			throws PulsarClientException {
-		Schema<T> schema = SchemaUtils.getSchema(message);
+		Schema<T> schema = this.schema != null ? this.schema : SchemaUtils.getSchema(message);
 		return this.producerFactory.createProducer(topic, schema, messageRouter, this.interceptors);
 	}
 
+	/**
```

Comment:
I am a bit torn on having this stored at the template level. Up until now we have made all inputs on the send methods. However, I also understand that adding params to that API increases the permutations of variants of the API methods along w/ maintenance/testing of them. This is probably a field that will not be changed on ever send call (but rather infrequently) so it is probably fine here at the top level. 

In general, the things I don't like about this pattern are:
1. keeping state makes the PT not thread-safe. Multiple threads could trample each other schema setting
2. as an API user it is not clear to me by looking at the `send` APIs that I should/may need to also call `setSchema` prior to invoking. Then once I find that info out I have to wonder what is it currently set to? Then should I set it back to what it was before I set it (aka cleanup after myself).

Reply:
We discussed offline and will explore a fluent send API in the near future. For now this approach is simple and works.